src/content/reference/cluster-upgrades/index.md

```diff
@@ -0,0 +1,187 @@
++++
+title = "Cluster Upgrades with Giant Swarm"
+description = "A detailed explanation how Kubernetes and other components are upgraded in a Giant Swarm installation"
+date = "2018-10-04"
+weight = 30
+type = "page"
++++
+
+# Cluster Upgrades with Giant Swarm
+
+A tenant cluster in a Giant Swarm installation is running a stack comprising many software components, provided by the Kubernetes project and other open source projects or software vendors, as well as by Giant Swarm.
+In order to keep all components up-to-date, to allow you to benefit from latest improvements, features and security fixes, we provide upgrades for the entire software stack in tenant clusters.
+
+At Giant Swarm we believe that frequent, small updates are a way to keep change in the system manageable.
+We invest a lot of effort in making this happen without disrupting your usage of your clusters.
+
+In this article, we explain how upgrades work in detail and how you should provide your workloads in order to keep them running during a tenant cluster upgrade and prevent disruptions of your applications.
+
+## Background and Concepts
+
+### The tenant cluster stack
+
+Among the third party components building a tenant cluster stack are
+
+- CoreOS Container Linux as the nodes' operating system
+- HashiCorp Vault as a public key infrastructure
+- Docker as a container runtime environment
+- CoreOS etcd as distributed storage for Kubernetes and Vault
+- Kubernetes with its many sub-components
+- Project Calico for virtual networking
+- CoreDNS for cluster-internal name resolution
+- NGINX Ingress Controller for connecting services with load balancers
+- Prometheus node exporter for hardware and OS metrics
+
+as well as many operators and controllers created and maintained by Giant Swarm.
+
+### Releases
+
+All of the items in the list above are released independent of each other by their vendors.
+At Giant Swarm we "bundle" specific versions of these components of the tenant cluster stack into a **release**.
+A release is identified by a [semantic version number](https://semver.org/).
+
+Once deployed, the tenant cluster stack is **immutable**.
+All components are deployed based on images, either of virtual machines or of Docker containers.
+No changes are ever made to components at runtime.
+This ensures that the stack running in your environment is the exact stack we have tested before.
+
+As a consequence, the only way to change the stack is to perform an upgrade, to switch to a new release.
+
```

Comment:
Maybe here we can add an exception. Right now chart operator installs the charts pushed by the cluster operator (which is bound to a version) but IIRC we will change it and then it would not be the case. WDYT?

Reply:
no, we won't change this, only for optional apps, which are not part of cluster and this process, those apps would be more like customer workloads, but that is for the future anyway